tests/playwright/src/model/pages/kubernetes-context-page.ts

```diff
@@ -0,0 +1,117 @@
+import { SettingsPage } from './settings-page';
+
+export class KubeContextPage extends SettingsPage {
+  readonly heading: Locator;
+  readonly content: Locator;
+  readonly contextTable: Locator;
+
+  constructor(page: Page) {
+    super(page, 'Kubernetes');
+    this.heading = this.page.getByLabel('Title', { exact: true });
+    this.content = this.page.getByLabel('Content');
+    this.contextTable = this.content.getByLabel('Contexts');
+  }
+
+  async pageIsEmpty(): Promise<boolean> {
+    const emptyHeading = this.page.getByRole('heading', { name: 'No Kubernetes contexts found', exact: true });
+    return (await emptyHeading.count()) > 0;
+  }
+
+  async getContextRowByName(name: string): Promise<Locator> {
+    return this.contextTable.getByLabel(name, { exact: true });
+  }
+
+  async isContextDefault(name: string): Promise<boolean> {
+    const row = await this.getContextRowByName(name);
+    if (row === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const bannerText = await row.getByLabel('Current Context').textContent();
+
+    return bannerText === 'Current Context';
+  }
+
+  async isContextReachable(name: string): Promise<boolean> {
+    const row = await this.getContextRowByName(name);
+    if (row === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const contextReachable = row.getByLabel('Context Reachable');
+
+    return (await contextReachable.count()) > 0;
+  }
+
+  async setDefaultContext(name: string): Promise<void> {
+    const contextRow = await this.getContextRowByName(name);
+    if (contextRow === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const switchButton = contextRow.getByLabel('Set as Current Context');
+    await playExpect(switchButton).toBeEnabled();
+    await switchButton.click();
+  }
+
+  async deleteContext(name: string, handleConfirmation: boolean = true): Promise<void> {
+    const contextRow = await this.getContextRowByName(name);
+    if (contextRow === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const deleteButton = contextRow.getByLabel('Delete Context');
+    await playExpect(deleteButton).toBeEnabled();
+    await deleteButton.click();
+    if (handleConfirmation) {
+      await handleConfirmationDialog(this.page, 'Delete Context');
+    }
+  }
+  async getContextName(context: string): Promise<string | null> {
+    const row = await this.getContextRowByName(context);
+    return row.getByLabel('Context Name').textContent();
+  }
+
+  async getContextCluster(context: string): Promise<string | null> {
+    const row = await this.getContextRowByName(context);
+    return row.getByLabel('Context Cluster').textContent();
+  }
+
+  async getContextServer(context: string): Promise<string | null> {
+    const row = await this.getContextRowByName(context);
+    return row.getByLabel('Context Server').textContent();
+  }
```

Comment:
```suggestion
  async getContextServer(context: string): Promise<string> {
    return await this.getContextRowByName(context).getByLabel('Context Server').innerText();
  }
```

using innerText you can avoid the null, since innerText only returns a string, it returns an empty string if the locator has no text.

you can make all the methods the same way.